10up/insecure-content-warning
# Fix preview button issue

### Description of the Change

If using the Classic Editor, if you click on the publish button and you have insecure warnings that show and you then click on the preview button, both buttons end up disabled and a loading icon shows up. Both issues persist until you refresh the page.

At a high level, WordPress doesn't account for the scenario of where someone would be able to click the publish button and then within the same session, click the preview button, as the publish button will by default cause a page refresh. In our case, if you have insecure assets, clicking the publish button stops publishing from happening but doesn't stop all the code WordPress has in place. This ends up causing issues if you then click the preview button.

I couldn't find a great way to handle this but this PR attempts to fix this by listening for clicks on the preview button and then checking to see if any insecure warnings are currently showing. If so, we can safely assume the publish button was clicked and so we re-enable as many things as possible. Also added a check so if the publish button is disabled, we don't run our scan

### Alternate Designs

None

### Benefits

Things will still work if using the Classic Editor and you first try to publish with insecure assets and then want to preview

### Possible Drawbacks

I didn't run into any issues but there could potentially be other things that WordPress disables or enables under this scenario that we haven't accounted for

### Verification Process

Before the update:
1. Open a post in the Classic Editor
2. Ensure this post has insecure items
3. Click the publish button and notice the warnings show
4. Click the preview button and notice a preview window opens
5. Go back to the post and notice the preview and publish buttons are disabled and a loading icon is shown

After the update:
1. Open a post in the Classic Editor
2. Ensure this post has insecure items
3. Click the publish button and notice the warnings show
4. Click the preview button and notice a preview window opens
5. Go back to the post and notice the preview and publish buttons are enabled and no loading icon is shown

### Checklist:


### Applicable Issues

Fixes #24
